frontend/src/views/ApplicationRegister.vue

```diff
@@ -127,52 +127,51 @@ export default {
     this.refreshForm()
   },
   methods: {
+    async fetchRecruitmentItems() {
+      const { data: recruitmentItems } = await RecruitmentApi.fetchItems(this.recruitmentId)
+      this.recruitmentItems = recruitmentItems.map(recruitmentItem => ({
+        ...recruitmentItem,
+        contents: "",
+      }))
+      return recruitmentItems
+    },
+    async fetchApplicationForm() {
+      const { data: applicationForm } = await ApplicationFormsApi.fetchForm({
+        token: this.$store.getters["token"],
+        recruitmentId: this.recruitmentId,
+      })
```

Comment:
```suggestion
    async fetchApplicationForm() {
      try {
      const { data: applicationForm } = await ApplicationFormsApi.fetchForm({
        token: this.$store.getters["token"],
        recruitmentId: this.recruitmentId,
      })
      } catch (e) {
      if (this.isEditing) {
          alert("저장된 지원서가 존재하지 않습니다.")
          this.$router.replace("/")
      }
```
각 메서드를 try ~ catch ~ 로 감싸면 404를 따로 체크하지 않아도 될 것 같아요.

어떤 api 콜에 대해서 어떤 처리를 해야하는지도 조금 더 분명해질 것 같고요.

Reply:
그래도 404는 체크해야 합니다. /new에서 지원서가 없는 경우 (정상) 와 진짜 에러가 난 경우를 구분하기 위해서죠.

Reply:
그리고 404 체크는 refreshForm에서 해야 하는 이유가 있는데 바로 /new에서 404 에러가 날 시 fetchApplicationForm에서 에러가 캐치된 채로 계속 진행하게 되면 fillForm에서 에러가 발생하기 때문입니다.

Reply:
/new는 `fetchForm`을 할 필요가 없지 않나요?

Reply:
fetchForm을 하지 않으면 이미 임시 저장된 지원서가 있는데도 /new로 접근하고 있지 않은지 여부를 알 수 없습니다. 이러한 기능이 필요없다면 지워도 되는데 필요하지 않을까요?

Reply:
이해를 잘못하고 있었군요. 감사합니다 :)

Reply:
메세지를 "잘못된 요청입니다"보다 "이미 지원한 모집입니다"라고 하는 것도 명확할 것 같아요

Reply:
해당 상황에 대한 메시지는 백엔드에서 보내주고 있습니다.
"잘못된 요청입니다"는 예측하지 못한 에러가 났을 때에 한정해서 표시되도록 하고 있습니다.

Reply:
멋진 이야기들이군요👍

Reply:
제가 코드를 주의 깊게 안봤군요 😂 감사합니다!